docs/bugs.json

```diff
@@ -1,4 +1,15 @@
 [
+    {
+        "uid": "SOL-2022-1",
+        "name": "AbiEncodeCallLiteralAsFixedBytesBug",
+        "summary": "Literals used in place of a fixed bytes argument in ``abi.encodeCall`` were encoded incorrectly",
+        "description": "The compiler only considered the types of the expressions in the second argument of ``abi.encodeCall`` itself, but not the parameter types of the function given as first argument. While in almost all cases the abi encoding of the type of the expression matches the abi encoding of the parameter type of the given function, since the type checker ensures the expression is implicitly convertible to the respective parameter type, this is not true for literals that are implicitly convertible to fixed bytes types shorter than 32 bytes, which have to be shifted to become left-aligned. This shifting was not performed.",
```

Comment:
```suggestion
        "description": "For the encoding, the compiler only considered the types of the expressions in the second argument of ``abi.encodeCall`` itself, but not the parameter types of the function given as first argument. In almost all cases the abi encoding of the type of the expression matches the abi encoding of the parameter type of the given function. This is because the type checker ensures the expression is implicitly convertible to the respective parameter type. However, this is not true for literal numbers that are implicitly convertible to fixed bytes types shorter than 32 bytes. They were encoded as numbers instead of being shifted to become left-aligned.",
```

Reply:
Rephrased somewhat following most of the suggestion, but also added the string literal case.

Reply:
Should we mention that ``very low`` is due to any single test on an occurrence of the bug will catch it?

Reply:
We never justified our severity ratings.